### ISSUE TITLE
Unable to Render Images from FlexForm Field within Collection in TYPO3 Frontend

### ISSUE THREAD

**Issue by tgrigoryan234** [FriendsOfTYPO3/content-blocks#292]
I have an image field within a FlexForm field, but in the frontend, only the count of images is loaded, not the actual images. To address this, I need to write a custom Processor. This becomes more complex when the FlexForm field is part of a collection within a content element.
Example Content Element config.yaml
....
fields:
  -
    identifier: teasers
    type: Collection
    minitems: 1
    table: tx_teasers
    labelField: title
    appearance:
      collapseAll: true
      levelLinksPosition: bottom
    fields:
      - identifier: teaser
        type: FlexForm
        label: Teaser Settings

The FlexForm (teaser) inside this collection includes an image field. How can I retrieve and display the actual image in the frontend?
There are currently no hooks or  signals or options in the Data Decoration process that allow adding a field to Content Block data. In the past, this was possible using the offsetSet function. I don’t understand why it was removed.

**Comment by nhovratov** [comment]
Which version are you on? TYPO3 v13 / v1 or TYPO3 v12 / 0.x?

**Comment by nhovratov** [comment]
In v13 FlexForm relations should be resolved as well. Could you please show me your complete yaml snippet of the FlexForm field?

**Comment by nhovratov** [comment]
Just tested it with this configuration and it works:

```
name: example/card-group
fields:
  - identifier: header
    useExistingField: true
  - identifier: settings
    type: FlexForm
    fields:
      - identifier: image
        type: File
  ```
  
  
![Screenshot 2024-11-14 at 10-10-21 Home](https://github.com/user-attachments/assets/d93473b3-efaf-46ca-b91f-93a5dbe57f0a)

**Comment by nhovratov** [comment]
OK, your are RIGHT. When putting the FlexForm inside a Collection it does indeed not work.

**Comment by nhovratov** [comment]
We will work on a bugfix in Core. Just one question: Why does the File field has to be inside FlexForm? Couldn't you just place it inside the Collection table?

> There are currently no hooks or signals or options in the Data Decoration process that allow adding a field to Content Block data. In the past, this was possible using the offsetSet function. I don’t understand why it was removed.

There is! But it's not documented yet. See: https://github.com/FriendsOfTYPO3/content-blocks/issues/278

The offset function was removed in favor of this Event. This Event has more flexibility and it's easier to work with when having nested Collections like in your case.

**Comment by tgrigoryan234** [comment]
I wanted to create a FlexForm structure, for example, for a teaser with the same fields that could be used both as a standalone element and within a slider or accordion. To achieve this, I created a teaser FlexForm in tt_content and reused it across various content blocks.

**Comment by nhovratov** [comment]
Created a forge issue: https://forge.typo3.org/issues/105597

**Comment by tgrigoryan234** [comment]
Thank you!

**Comment by nhovratov** [comment]
This is fixed in Core now: https://review.typo3.org/c/Packages/TYPO3.CMS/+/87040

Will be fixed with release of TYPO3 13.4.2

**Comment by nhovratov** [comment]
TYPO3 13.4.2 was released today.
